pkg/output/printer_factory.go

```diff
@@ -0,0 +1,81 @@
+package output
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	stackservice "github.com/chanzuckerberg/happy/pkg/stack_mgr"
+	"github.com/chanzuckerberg/happy/pkg/util"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v3"
+)
+
+type Printer interface {
+	PrintStacks(stackInfos []stackservice.StackInfo) error
+	Fatal(err error)
+}
+
+type TextPrinter struct{}
+type JSONPrinter struct{}
+type YAMLPrinter struct{}
+
+func NewPrinter(outputFormat string) Printer {
+	switch outputFormat {
+	case "json":
+		return &JSONPrinter{}
+	case "yaml":
+		return &YAMLPrinter{}
+	default:
+		return &TextPrinter{}
+	}
+}
+
+func (p *TextPrinter) PrintStacks(stackInfos []stackservice.StackInfo) error {
+	headings := []string{"Name", "Owner", "Tags", "Status", "URLs", "LastUpdated"}
```

Comment:
not for this PR, but I think we should change the table printer to take advantage of the `header` struct tags like https://github.com/lensesio/tableprinter instead of us hard-coding these. gives us more flexibility and we don't need type specific print functions

Reply:
Created a ticket: https://czi-tech.atlassian.net/browse/CCIE-344